lua/kznllm-v3/specs/openai.lua

```diff
@@ -64,21 +64,21 @@ M.VLLMProvider = M.OpenAIProvider:new {
 ---@return string
 function M.OpenAIProvider:handle_sse_stream(line)
   -- based on sse spec (OpenAI spec uses data-only server-sent events)
-  local data = line:match '^data: (.+)$'
+  local content = {}
 
-  local content = ''
-
-  if data and data:match '"delta":' then
-    local json = vim.json.decode(data)
-    -- sglang server returns the role as one of the events and it becomes `vim.NIL`, so we have to handle it here
-    if json.choices and json.choices[1] and json.choices[1].delta and json.choices[1].delta.content and json.choices[1].delta.content ~= vim.NIL then
-      content = json.choices[1].delta.content
-    else
-      vim.print(data)
+  for _, d in ipairs(vim.split(line, 'data: ')) do
+    if d and d:match '"delta":' then
+      local json = vim.json.decode(d)
+      -- sglang server returns the role as one of the events and it becomes `vim.NIL`, so we have to handle it here
+      if json.choices and json.choices[1] and json.choices[1].delta and json.choices[1].delta.content and json.choices[1].delta.content ~= vim.NIL then
+        table.insert(content, json.choices[1].delta.content)
+      else
+        vim.print(d)
+      end
     end
   end
 
-  return content
+  return table.concat(content, '')
```

Comment:
ok well if i go off SSE spec, you can send multiple consecutive lines

> When the EventSource receives multiple consecutive lines that begin with data:, [it concatenates them](https://html.spec.whatwg.org/multipage/#dispatchMessage), inserting a newline character between each one. Trailing newlines are removed.

```
---
data: {"id":"fd57afc8-864a-43c8-97c1-30f9af9b1baf","object":"chat.completion.chunk","created":1732370382,"model":"deepseek-chat","system_fingerprint":"fp_1c141eb703","choices":[{"inde
x":0,"delta":{"role":"assistant","content":""},"logprobs":null,"finish_reason":null}]}
data: {"id":"fd57afc8-864a-43c8-97c1-30f9af9b1baf","object":"chat.completion.chunk","created":1732370382,"model":"deepseek-chat","system_fingerprint":"fp_1c141eb703","choices":[{"inde
x":0,"delta":{"content":"To"},"logprobs":null,"finish_reason":null}]}
---
data: {"id":"fd57afc8-864a-43c8-97c1-30f9af9b1baf","object":"chat.completion.chunk","created":1732370382,"model":"deepseek-chat","system_fingerprint":"fp_1c141eb703","choices":[{"inde
x":0,"delta":{"content":" implement"},"logprobs":null,"finish_reason":null}]}
---
data: {"id":"fd57afc8-864a-43c8-97c1-30f9af9b1baf","object":"chat.completion.chunk","created":1732370382,"model":"deepseek-chat","system_fingerprint":"fp_1c141eb703","choices":[{"inde
x":0,"delta":{"content":" the"},"logprobs":null,"finish_reason":null}]}
---
data: {"id":"fd57afc8-864a-43c8-97c1-30f9af9b1baf","object":"chat.completion.chunk","created":1732370382,"model":"deepseek-chat","system_fingerprint":"fp_1c141eb703","choices":[{"inde
x":0,"delta":{"content":" Fib"},"logprobs":null,"finish_reason":null}]}
---
data: {"id":"fd57afc8-864a-43c8-97c1-30f9af9b1baf","object":"chat.completion.chunk","created":1732370382,"model":"deepseek-chat","system_fingerprint":"fp_1c141eb703","choices":[{"inde
x":0,"delta":{"content":"onacci"},"logprobs":null,"finish_reason":null}]}
---
data: {"id":"fd57afc8-864a-43c8-97c1-30f9af9b1baf","object":"chat.completion.chunk","created":1732370382,"model":"deepseek-chat","system_fingerprint":"fp_1c141eb703","choices":[{"inde
x":0,"delta":{"content":" sequence"},"logprobs":null,"finish_reason":null}]}
```

Reply:
This actually makes the anthropic spec logic more correct because events come together with data as intended and we don't have to store any state

```
haiku-3-5-chat              claude-3-5-haiku-20241022 | temp = 0.7
---
event: message_start
data: {"type":"message_start","message":{"id":"msg_01RXrj8hgRc7yNZ7gjpwrr4C","type":"message","role":"assistant","model":"claude-3-5-haiku-20241022","content":[],"stop_reason":null,"s
top_sequence":null,"usage":{"input_tokens":1808,"cache_creation_input_tokens":0,"cache_read_input_tokens":0,"output_tokens":2}}       }
event: content_block_start
data: {"type":"content_block_start","index":0,"content_block":{"type":"text","text":""}      }
event: ping
data: {"type": "ping"}
event: content_block_delta
data: {"type":"content_block_delta","index":0,"delta":{"type":"text_delta","text":"I'll"}        }
---
event: content_block_delta
data: {"type":"content_block_delta","index":0,"delta":{"type":"text_delta","text":" analyze"}           }
---
event: content_block_delta
data: {"type":"content_block_delta","index":0,"delta":{"type":"text_delta","text":" the implementation"}              }
```